### PR TITLE
Table managementv2

### PR DESCRIPTION
**Pull request by ScottyKnows44**
Square, Circle, Rectangle Components with their stories for Storybook

### CODE REVIEW

File: src/Containers/CircleTable.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+           numOfChairs = 4,
+           partyName = 'Null',
+           occupancyStatus = occupancyStatusTypes.Vacant,
+           reservationTime = Date.now(),
+           ...props
+       }) => {
+    const {colors} = useTheme();
+
+    /**
+     * Primary function that returns the color from the occupancyStatus
```
Comment (ralph-dev) [inline src/Containers/CircleTable.tsx:47]: This is not proper JSDOC the parameter is not typed

File: src/Containers/CircleTable.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            </TableBody>
+        </div>
+    );
+};
+
+    /**
+    * variables for the styled components
+     */
+
+    const TableBody = styled.div`
```
Comment (ralph-dev) [inline src/Containers/CircleTable.tsx:92]: Why is there random indents here?

File: src/Containers/CircleTable.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            margin-left: auto;
+            margin-right: auto;
+            margin-top: 3rem;
+        `;
+
+    const TextWhiteDiv = styled.div`
+            color: #fff;
+        `;
+    const TextOccupancyColor = styled.div`
+           //color: ${({theme}) => theme.colors.primary};
```
Comment (ralph-dev) [inline src/Containers/CircleTable.tsx:145]: Remove commented code

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Needs improvement

File: src/Containers/RectangleTable.tsx [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+    function getChairs(numOfChairs: number){
+
+        if(numOfChairs < 1 || numOfChairs > 6 ){
+            return 2;
+        }
+
+        if(numOfChairs < 3){
+            return 2;
+        } else if (numOfChairs < 5){
+            return 4;
```
Comment (ralph-dev) [inline src/Containers/RectangleTable.tsx:63]: These are magic numbers what does it mean

File: src/Containers/CircleTable.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import styled, {useTheme} from "styled-components";
+
+/**
+ * Primary UI component for user interaction
+ */
+export interface ICircleTable { /**
```
Comment (ralph-dev) [inline src/Containers/CircleTable.tsx:7]: This /** should be on the next line

**Review by ralph-dev — COMMENTED** [review]
Small addition

File: debug.log [hunk 6]
```diff
@@ -0,0 +1 @@
+[1111/150741.820:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
```
Comment (ralph-dev) [inline debug.log:1]: add this to gitignore

File: src/Containers/Chair.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+         */
+        function chairSwitch(): JSX.Element
+        {
+            switch (position) {
+            case 'top':
+                return <TopChair />;
+            case 'bottom':
+                return <BottomChair />;
+            case 'left':
+                return <div><LeftChair /></div>;
```
Comment (ralph-dev) [inline src/Containers/Chair.tsx:42]: why is there an extra div here?

File: src/Containers/Chair.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+            case 'right':
+                return <RightChair />;
+            default:
+                return <div />;
```
Comment (ralph-dev) [inline src/Containers/Chair.tsx:46]: why not return null?

File: src/Containers/Chair.tsx [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+const TopChair=styled.div`
+
+    border-top-left-radius: 3rem;
+    border-top-right-radius: 3rem;
+    height: 2rem;
+    width: 10rem;
+    margin-bottom: 0.25rem;
+    margin-left: auto;
+    margin-right: auto;
+    background-color: #6c757d;
```
Comment (ralph-dev) [inline src/Containers/Chair.tsx:65]: use theme object

File: src/Containers/Chair.tsx [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    border-bottom-right-radius: 3rem;
+    width: 2rem;
+    height: 10rem;
+    margin-top:auto;
+    margin-bottom: auto;
+    margin-left: 1.25rem;
+    background-color: #6c757d;
+`;
+
+const BottomChair=styled.div`
```
Comment (ralph-dev) [inline src/Containers/Chair.tsx:93]: do BaseChair = styled.div``;
then do 
const RightChair = styled(BaseChair)``;
to reduce duplicate styles

**Review by ralph-dev — APPROVED** [review]
LGTM

File: src/Containers/Chair.tsx [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+        position = 'top',
+        isSeated = false,
+        ...props
+    }) => {
+
+        /**
+         * Returns a JSX element for the Chair with the correct styles
+         * @returns {JSX.Element} the correct JSX.Element based on position
+         */
+        function chairSwitch(): JSX.Element
```
[the remark this answers is not in the document]
Comment (rogers-ch) [reply src/Containers/Chair.tsx]: Ralph, I updated this with ES6 syntax and solved the conflict in the index.ts file. I can merge the pull request if this looks good to you.

File: src/Containers/Chair.tsx [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+    }) => {
+
+        const {colors} = useTheme();
+
+        /**
+         * Returns a JSX element for the Chair with the correct styles
+         * @returns {JSX.Element} the correct JSX.Element based on position,
+         * or null if no position is provided
+         */
+        const chairSwitch: () => (JSX.Element | null) = () =>
```
Comment (ralph-dev) [inline src/Containers/Chair.tsx:37]: Don't define inline types btw :)

**Pull request merged**